src/main/scala/is/hail/io/vcf/ExportVCF.scala

```diff
@@ -150,27 +150,33 @@ object ExportVCF {
     toAppend.foreachBetween(sb.append(_))(sb += ',')
   }
 
-  def writeGenotype(sb: StringBuilder, sig: TStruct, fieldOrder: Array[Int], a: Annotation) {
-    val r = a.asInstanceOf[Row]
+  def writeGenotype[T](sb: StringBuilder, sig: TStruct, fieldOrder: Array[Int], g: Row) {
     val fields = sig.fields
-    assert(r.length == fields.length, "annotation/type mismatch")
+    assert(g.length == fields.length, "annotation/type mismatch")
 
     fieldOrder.foreachBetween { i =>
-      emitFormatField(fields(i), sb, r.get(i))
+      emitFormatField(fields(i), sb, g.get(i))
     }(sb += ':')
   }
 
   def writeGenotype(sb: StringBuilder, g: Genotype) {
-    sb.append(g.gt.map { gt =>
+    if (g == null) {
+      sb.append("./.")
```

Comment:
does this compile to the same thing as if we saved this "./." string as a val somewhere on a static object?